src/ate/ate_dll.cc

```diff
@@ -283,7 +283,7 @@ int TokensCopy(size_t count, const pa::DeriveTokensResponse &resp,
     }
 
     resp_token.size = token.size();
-    memcpy(resp_token.data, token.c_str(), sizeof(resp_token.data));
+    memcpy(resp_token.data, token.c_str(), resp_token.size);
```

Comment:
Why `c_str()` here?  The `size` doesn't include the `nul` terminator and `c_str` provides a pointer to a buffer that will provide it.  I think `data()` is more appropriate here.

We currently have error checking to ensure that the `token.size()` is not larger than the target buffer (`resp_token.data`).  Is it possible that `token.size()` might be too shorter than `sizeof(resp_token.data)`, and if so:
- What is the appropriate data pattern for the rest of `resp_token.data`?
- Who is responsible for setting or initializing that portion of `resp_token.data`?